pkg/csi/cinder/controllerserver.go

```diff
@@ -392,6 +475,78 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		klog.Errorf("Failed to WaitSnapshotReady: %v", err)
 		return nil, status.Errorf(codes.Internal, "CreateSnapshot failed with error %v", err)
 	}
+	// If snapshotType is 'backup', create a backup from the snapshot and delete the snapshot.
```

Comment:
it makes the backup process robust. Lets say that we have 5TB cinder volume, you will take "cinder backup" from it. It means that it will copy entire 5TB volume from storage system A to storage system B by using external network. It will take a lot of time. During copying the content of the volume can change a lot and it means that in some situations it will corrupt the backup. Usually when taking snapshot from existing volume it means that it can be really snapshot if the storage system supports that (taking snapshot inside same volume backend can support that, but not all support that still..). Which means that the possibility to changes and issues are a lot smaller.